### PR TITLE
Reorganize parsing

### PR DESCRIPTION
**Pull request by natario1**
This PR adds no new functionality, it's just code pasted in different classes.

I removed `BranchResponseParser` (which was parsing only Search results, not QH/AS) and split its code into the classes that are being parsed: `BranchAppResult` and `BranchLinkResult`.

Now parsing is consistent and all user facing POJOs (BranchAppResult, BranchLinkResult, BranchQueryHintResult, BranchAutoSuggestResult, BranchSearchResult, ...) have a static `createFromJson()` function that is used when parsing.

This helps with splitting responsibilities and prepares the ground for link handling v2.

### CODE REVIEW

File: BranchSearchSDK/src/main/java/io/branch/search/BranchSearchResult.java [hunk 1]
```diff
@@ -40,4 +52,35 @@ public BranchSearchRequest getBranchSearchRequest() {
...
+                                             @NonNull JSONObject json) {
+        String correctedQuery = null;
+        if (json.has(KEY_CORRECTED_QUERY)) {
+            correctedQuery = json.optString(KEY_CORRECTED_QUERY);
+        }
+        List<BranchAppResult> results = new ArrayList<>();
+        if (json.optBoolean(KEY_SUCCESS)) {
+            JSONArray resultsJson = json.optJSONArray(KEY_RESULTS);
+            if (resultsJson != null) {
+                String requestId = Util.optString(json, KEY_REQUEST_ID);
```
Comment (bklastaitis-branch) [inline BranchSearchSDK/src/main/java/io/branch/search/BranchSearchResult.java:73]: It doesn't seem like `BranchAppResult.createFromJson` uses `requestId`, plus we never check whether it's `null` even though `BranchAppResult.createFromJson` annotates that params as NonNull. Can probably remove the parsing of `requestId`/passing it to `BranchAppResult.createFromJson`

Comment (natario1) [reply BranchSearchSDK/src/main/java/io/branch/search/BranchSearchResult.java]: I have removed this!

File: BranchSearchSDK/src/main/java/io/branch/search/BranchAppResult.java [hunk 2]
```diff
@@ -141,6 +157,7 @@ public BranchSearchError openApp(Context context, boolean fallbackToPlayStore) {
      * @return BranchSearchError {@link BranchSearchError} object to pass any error with complete action. Null if succeeded.
      * @deprecated the search deep link, if available, will be among other link results for this app
      */
+    @Deprecated
```
[the remark this answers is not in the document]
Comment (natario1) [reply BranchSearchSDK/src/main/java/io/branch/search/BranchAppResult.java:160]: Yeah I wanted to keep the annotation as we use it in all other deprecated methods as well.

**Comment by bklastaitis-branch** [comment]
👍

**Pull request merged**